xla/service/gpu/gpu_performance_model.h

```diff
@@ -39,6 +51,72 @@ class GpuPerformanceModel {
   // Writes estimated execution time to FusionBackendConfig.reification_cost.
   static void RecordEstimatedRunTime(HloInstruction* instruction,
                                      const GpuHloCostAnalysis* cost_analysis);
+  static absl::Duration ComputeTime(const GpuDeviceInfo& gpu_device_info,
+                                    int64_t n_flops, int64_t n_threads);
+
+  static absl::Duration ProducerInputAccessTime(
+      const GpuHloCostAnalysis* cost_analysis,
+      const GpuDeviceInfo& gpu_device_info, const HloInstruction* producer,
+      const HloInstruction* fused_consumer = nullptr);
+};
+
+class GpuPerformanceWithCollectiveModel : public GpuPerformanceModel {
+ public:
+  // Different algorithms that can be used to perform the collective.
+  enum CollectiveAlgo {
+    RING = 0,
+    TREE,
+  };
+
+  // Table for max system bandwidths.
+  // Index 1 is for well-known GPU link BWs, used for intra-node estimate.
+  static constexpr double low_latency_max_bandwidths[3] = {
+      39.0 /* Volta-N1 */, 87.7 /* Ampere-N1 */, 87.7 /* Hopper-N1 */
```

Comment:
Would be good to document the source of these numbers. Whats does N1 stand for? A quick Google search says https://images.nvidia.com/content/volta-architecture/pdf/volta-architecture-whitepaper.pdf NVLink BW is 300 Gb/s and signaling rate is 25 Gigabits/sec. Also clarifying the unit here would be great

Reply:
This is actually NCCL's internal parameters for its low latency algorithm. This is an estimated value based on unidirectional bandwidth per channel(cta) plus a discounted factor of 95% since it will not be able to use peak bandwidth due to temp buffer allocation and send recv synchronization etc. I have updated the comment with units and removed N1(single-node) from notation.